combine meshgrid native functions

dagitses opened the pull request:
Stack created with [Sapling](https://sapling-scm.com). Best reviewed with [ReviewStack](https://reviewstack.dev/pytorch/pytorch/pull/93092).
* __->__ #93092

combine meshgrid native functions

Summary: The FC window has long passed, no need to keep them distinct.

Test Plan: Rely on CI.

dagitses commented:
@ezyang is the check right? Does this break BC?

ezyang commented on aten/src/ATen/native/TensorShape.cpp:
```diff
@@ -3527,7 +3526,7 @@ std::vector<Tensor> meshgrid(TensorList tensors,
   bool swap_first_and_second_tensors = false;
 
-  if (indexing == "xy") {
+  if (*indexing == "xy") {
```
nit: would be better to rename the optional `maybe_indexing` and have a non optional `indexing`

dagitses replied:
done!

ezyang commented:
what about the deprecation?

Commit:
combine meshgrid native functions

Summary: The FC window has long passed, no need to keep them distinct.

Test Plan: Rely on CI.

dagitses commented:
> what about the deprecation?



> what about the deprecation?



> what about the deprecation?

the warning is still there. Should I just complete the deprecation after the next release is cut?

ezyang commented:
oh ok i could believe we haven't waited long enough. carry on
